### PR TITLE
adding marslab recipe

### PR DESCRIPTION
**Pull request by Sierra-MC**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Sierra-MC** [comment]
@conda-forge/staged-recipes please review.

File: recipes/marslab/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    - marslab
+  commands:
+    # removed pip check due to discrepancies with clize metadata in PyPI vs conda-forge
+  requires:
+    - pip
+
+about:
+  home: https://github.com/millionconcepts/marslab.git
+  summary: Utilities for working with observational data of Mars.
+  description: marslab is the core support library for the marslab ecosystem (https://github.com/MillionConcepts/marslab-reference). It defines common formats, shared references, and support objects, many of which are also useful on their own. Highlights include Look, a lightweight construction kit for multispectral visualizations; and BandSet, an object that streamlines operations on multispectral observations, treating entire clusters of related images as coherent wholes.
```
Comment (ocefpaf) [inline recipes/marslab/meta.yaml:53]: Do you mind breaking this with new lines?

File: recipes/marslab/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+test:
+  imports:
+    - marslab
+  commands:
+    # removed pip check due to discrepancies with clize metadata in PyPI vs conda-forge
+  requires:
+    - pip
+
+about:
+  home: https://github.com/millionconcepts/marslab.git
```
Comment (ocefpaf) [inline recipes/marslab/meta.yaml:51]:
```suggestion
  home: https://github.com/millionconcepts/marslab
```

File: recipes/marslab/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    - dustgoggles
+    - fs
+    - hypothesis
+    - matplotlib-base
+    - more-itertools
+    - numpy
+    - pandas
+    - pathos
+    - pdr
+    - pytest
```
Comment (synapticarbors) [inline recipes/marslab/meta.yaml:35]: I don't think `pytest` and other packages needed only for testing should be included here. The list should mainly be what's defined in:
https://github.com/MillionConcepts/marslab/blob/main/setup.py#L19-L29

and then it's a choice whether to include other extra dependencies.

File: recipes/marslab/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    - python-dateutil
+    - python-Levenshtein
+    - scipy
+    - sympy
+
+test:
+  imports:
+    - marslab
+  commands:
+    # removed pip check due to discrepancies with clize metadata in PyPI vs conda-forge
```
Comment (synapticarbors) [inline recipes/marslab/meta.yaml:46]: Also, I'm unclear as to why `pip check` is causing issues when here and upstream there is no version restrictions on the dependencies.

File: recipes/marslab/meta.yaml (hunk 1 above)
Comment (Sierra-MC) [reply recipes/marslab/meta.yaml]: added /n characters to break this up.

**Comment by synapticarbors** [comment]
Instead of using newline characters you can use indentation and the pipe as shown in the example recipe:
https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L69-L73

File: recipes/marslab/meta.yaml (hunk 3 above)
Comment (Sierra-MC) [reply recipes/marslab/meta.yaml]: Removed pytest and pytest-cov

**Comment by Sierra-MC** [comment]
I believe I have addressed the team's concerns and have repassed all the linting and CI checks. Please let me know if there are any further steps before merge.

File: recipes/marslab/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    - matplotlib-base
+    - more-itertools
+    - numpy
+    - pandas
+    - pathos
+    - pdr
+    - python-dateutil
+    - python-Levenshtein
+    - scipy
+    - sympy
```
Comment (ocefpaf) [inline recipes/marslab/meta.yaml:38]: According to [upstream metadata](https://github.com/MillionConcepts/marslab/blob/v0.9.10/setup.py#L19-L28) these are the base requirements:

```suggestion
    - numpy
    - scipy
    - dustgoggles
    - fs
    - clize
    - pandas
    - more-itertools
    - pathos
    - cytoolz
```

You can add [some of the extras](https://github.com/MillionConcepts/marslab/blob/v0.9.10/setup.py#L30) but I think that adding the tests one are not necessary for the average user.

Comment (Sierra-MC) [reply recipes/marslab/meta.yaml]: Apologies I did not realize I had left another test package in there. I've now removed hypothesis. We prefer to keep the extras in due to the proficiency level of our typical user.

**Pull request merged**